packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from 'react';
+import { makeStyles } from '@mui/styles';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import FormControl from '@mui/material/FormControl';
+import Button from '@mui/material/Button';
+import DialogTitle from '@mui/material/DialogTitle';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+// import DialogContentText from '@mui/material/DialogContentText';
+
+import DialogActions from '@mui/material/DialogActions';
+import { withApplicationContext } from '../contexts/Application';
+
+const useStyles = makeStyles(_ => ({
+  centeredText: {
+    textAlign: 'center',
+  },
+  dialog: {
+    minWidth: 240,
+  },
+}));
+
+const ConnectionSettingsDialog = ({
+  onClose,
+  open,
```

Comment:
I was going to ask that this be `isOpen` to make clear it's declarative and not imperative but I see MUI's `Dialog` uses `open` so agree it's better to be consistent